pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
Yeah args is what you want to use when you have concerns about quoting,
shellScript is what you use when you want Jenkins like behavior. UI can
give you choice of args/command or shellScript.
On Feb 1, 2016 8:58 PM, "Clayton Coleman" <notifications@github.com> wrote:

> In pkg/build/api/v1beta3/types.go
> <https://github.com/openshift/origin/pull/6715#discussion_r51471192>:
>
> > @@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
> >  	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
> >  }
> >
> > +// BuildPostCommitSpec holds a build post commit hook specification.
> > +type BuildPostCommitSpec struct {
> > +	// Shell is an action that executes a shell script in a container
> > +	// running the build output image.
> > +	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
>
> I think we should be a bit more explicit about the variable name - Script
> or ShellScript is probably more specific and clearer than just Shell. Is
> Script more familiar to users or Shell to identify a chunk of something
> they place into the hook?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6715/files#r51471192>.
>